jacobian_follower/src/jacobian_follower/jacobian_follower.cpp

```diff
@@ -47,120 +102,140 @@ JacobianFollower::JacobianFollower(std::string const robot_namespace, double con
       world_frame_("robot_root"),
       robot_frame_(model_->getRootLinkName()),
       translation_step_size_(translation_step_size),
-      minimize_rotation_(minimize_rotation) {
+      minimize_rotation_(minimize_rotation),
+      collision_check_(collision_check),
+      visualize_(visualize),
+      robot_namespace_(robot_namespace) {
   if (!ros::isInitialized()) {
     ROS_FATAL_STREAM_NAMED(LOGGER_NAME,
                            "You must call ros::init before using JacobianFollower. "
                                << "If you're calling this from python, use arc_utilities.ros_init.rospy_and_cpp_init");
   }
   nh_ = std::make_shared<ros::NodeHandle>();
-  auto const waypoints_topic = ros::names::append(robot_namespace, "jacobian_waypoints");
-  vis_pub_ =
-      std::make_shared<ros::Publisher>(nh_->advertise<visualization_msgs::MarkerArray>(waypoints_topic, 10, true));
+  auto const interpolated_points_topic = ros::names::append(robot_namespace, "jacobian_waypoint_interpolated");
+  vis_pub_ = std::make_shared<ros::Publisher>(
+      nh_->advertise<visualization_msgs::MarkerArray>(interpolated_points_topic, 10, true));
 
-  auto const scene_topic = ros::names::append(robot_namespace, "move_group/monitored_planning_scene");
-  auto const service_name = ros::names::append(robot_namespace, "get_planning_scene");
-  scene_monitor_->startSceneMonitor(scene_topic);
-  scene_monitor_->requestPlanningSceneState(service_name);
   auto const display_planned_path_topic = ros::names::append(robot_namespace, "jacobian_follower_planned_traj");
   visual_tools_.loadTrajectoryPub(display_planned_path_topic, false);
   auto const display_robot_state_topic = ros::names::append(robot_namespace, "jacobian_follower_robot_state");
   visual_tools_.loadRobotStatePub(display_robot_state_topic, false);
+
+  if (collision_check_) {
```

Comment:
duplicated information. Seems that member variable `bool collision_check_` is uniquely determined by `constraint_fn`. In code you check `if(constraint_fn)`. This could cause unexpected behavior if a user sets `collision_check_` to false, but the code checks the (prevously set) `if(constraint_fn)`

Reply:
that can't happen because collision_check_ is set via the constructor. But you're right in that constraint_fn being default constructed is the same as `collision_check_=False`. I think the solution here is to make `collision_check_` not a member, just use the local variable that is the argument to the constructor